settings.ts

```diff
@@ -216,44 +226,31 @@ export class VoiceNotesSettingTab extends PluginSettingTab {
           })
       );
 
-    new Setting(containerEl)
-      .setName('Debug Mode')
-      .setDesc('Enable debug mode for additional logging')
-      .addToggle((toggle) =>
-        toggle.setValue(this.plugin.settings.debugMode).onChange(async (value) => {
-          this.plugin.settings.debugMode = value;
-          await this.plugin.saveSettings();
-        })
-      );
-
-    new Setting(containerEl)
-      .setName('Sync Interval')
-      .setDesc('Interval in minutes for automatic synchronization')
-      .addText((text) =>
-        text
-          .setPlaceholder('30')
-          .setValue(String(this.plugin.settings.syncInterval))
-          .onChange(async (value) => {
-            const interval = parseInt(value, 10);
-            if (!isNaN(interval) && interval > 0) {
-              this.plugin.settings.syncInterval = interval;
-              await this.plugin.saveSettings();
-            }
-          })
-      );
-
     new Setting(containerEl)
       .setName('Custom Note Template')
-      .setDesc('Custom template for synced notes. Available variables: {{title}}, {{date}}, {{transcript}}, {{audio_link}}, {{summary}}, {{tidy}}, {{points}}, {{todo}}, {{email}}, {{custom}}')
-      .addTextArea((text) =>
+      .setDesc('Custom template for synced notes. Available variables: {{title}}, {{date}}, {{transcript}}, {{audio_link}}, {{summary}}, {{tidy}}, {{points}}, {{todo}}, {{email}}, {{tweet}}, {{blog}}, {{custom}}, {{parent_note}} and {{related_notes}} (uncheck to add a custom frontmatter)')
+      .addTextArea((text) => {
         text
           .setPlaceholder(this.plugin.settings.noteTemplate)
           .setValue(this.plugin.settings.noteTemplate)
           .onChange(async (value) => {
             this.plugin.settings.noteTemplate = value;
             await this.plugin.saveSettings();
-          })
-      );
+          });
+        // Add autoresize to the textarea
+        text.inputEl.classList.add('autoresize');
+        autoResizeTextArea(text.inputEl);
+        text.inputEl.addEventListener('input', () => autoResizeTextArea(text.inputEl));
+        containerEl.appendChild(text.inputEl);
+      })
+      .addToggle((toggle) => {
+        toggle
+          .setValue(this.plugin.settings.useDefaultFrontmatter)
+          .onChange(async (value) => {
```

Comment:
Again, the purpose of this change is to respond to https://github.com/mysticcoders/voicenotes-sync/issues/40:
- If the option is set to True, the frontmatter used is the default frontmatter, with recording_id, duration, created_at and updated_at.
- Otherwise, the frontmatter does not exist (the user is therefore free to customise it in their jinja2 template). Only the recording_id property will be added for technical reasons.

_It is also possible to include the default frontmatter in the basic Jinja2 template, but I find that less clear and less practical than simply clicking on the option ! 🦦 _